format/FlightSql.proto

```diff
@@ -1549,6 +1658,17 @@ message DoPutUpdateResult {
   int64 record_count = 1;
 }
 
+/*
+ * Explicitly cancel a running query.
+ *
+ * For distributed queries (ones that use GetFlightInfo->DoGet), this lets a
+ * single client explicitly cancel work across all clients, given server support.
+ */
+message ActionCancelQuery {
```

Comment:
does there need to be a response associated with this?

Reply:
What would it contain?

Reply:
Could state.  CANCELLED, NOT_CANCELLABLE, ALREADY_DONE, CANCELLING.  Not sure this is useful, so feel free to ignore.

Reply:
If we declare the command idempotent, that could be useful. I suppose the server could just synchronously block until the query is cancelled but then it'd be unclear how to recover from a transient failure.